lib/config/validation.spec.ts

```diff
@@ -920,5 +921,56 @@ describe('config/validation', () => {
         },
       ]);
     });
+
+    it('errors if unallowed header in hostRules', async () => {
+      GlobalConfig.set({ allowedHeader: ['X-*'] });
+
+      const config = {
+        hostRules: [
+          {
+            matchHost: 'https://domain.com/all-versions',
+            header: {
+              'X-Auth-Token': 'token',
+              unallowedHeader: 'token',
+            },
+          },
+        ],
+      };
+      const { warnings, errors } =
+        await configValidation.validateConfig(config);
+      expect(warnings).toHaveLength(0);
+      expect(errors).toMatchObject([
+        {
+          message:
+            "hostRules header `unallowedHeader` is not permitted by this bot's `allowedHeader`.",
+          topic: 'Configuration Error',
+        },
+      ]);
+    });
+
+    it('errors if header values are not string', async () => {
+      GlobalConfig.set({ allowedHeader: ['X-*'] });
+
+      const config = {
+        hostRules: [
+          {
+            matchHost: 'https://domain.com/all-versions',
+            header: {
+              'X-Auth-Token': 10,
+            } as unknown as Record<string, string>,
+          },
+        ],
+      };
+      const { warnings, errors } =
+        await configValidation.validateConfig(config);
+      expect(warnings).toHaveLength(0);
+      expect(errors).toMatchObject([
+        {
+          message:
+            'Invalid hostRules header value configuration: should be a string.',
```

Comment:
```suggestion
            'Invalid hostRules header value configuration: header must be a string.',
```

How about this instead? Now it's clear the user `must` do something, and it's clear _where_ the user must change something: to the `header`.